### PR TITLE
Add warehouse information functionality

### PR DESCRIPTION
**Pull request by jeffreytjs**

### CODE REVIEW

File: src/main/java/seedu/clinic/logic/commands/AddCommand.java [hunk 1]
```diff
@@ -3,65 +3,97 @@
...
-            + "Parameters: "
-            + PREFIX_NAME + "NAME "
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Adds a supplier/warehouse to CLI-nic.\n"
+            + "Supplier Parameters: "
+            + PREFIX_SUPPLIER_NAME + "NAME "
             + PREFIX_PHONE + "PHONE "
             + PREFIX_EMAIL + "EMAIL "
-            + PREFIX_ADDRESS + "ADDRESS "
-            + "[" + PREFIX_TAG + "TAG]...\n"
+            + PREFIX_REMARK + "REMARK "
```
Comment (qlchan24) [inline src/main/java/seedu/clinic/logic/commands/AddCommand.java]: I think there should be a \n here

File: src/main/java/seedu/clinic/logic/commands/AddCommand.java [hunk 2]
```diff
@@ -3,65 +3,97 @@
...
             + PREFIX_EMAIL + "johnd@example.com "
-            + PREFIX_ADDRESS + "311, Clementi Ave 2, #02-25 "
-            + PREFIX_TAG + "friends "
-            + PREFIX_TAG + "owesMoney";
+            + PREFIX_REMARK + "Largest contractor \n"
+            + "Warehouse Parameters: "
+            + PREFIX_WAREHOUSE_NAME + "NAME "
+            + PREFIX_PHONE + "PHONE "
+            + PREFIX_ADDRESS + "Address "
+            + PREFIX_REMARK + "REMARK "
```
Comment (qlchan24) [inline src/main/java/seedu/clinic/logic/commands/AddCommand.java]: same as above

File: src/main/java/seedu/clinic/logic/commands/AddCommand.java [hunk 3]
```diff
@@ -3,65 +3,97 @@
...
 
-        if (model.hasSupplier(toAdd)) {
-            throw new CommandException(MESSAGE_DUPLICATE_SUPPLIER);
+        if (supplierToAdd != null) {
+            if (model.hasSupplier(supplierToAdd)) {
+                throw new CommandException(MESSAGE_DUPLICATE_SUPPLIER);
+            }
+            model.addSupplier(supplierToAdd);
+            return new CommandResult(String.format(MESSAGE_SUPPLIER_SUCCESS, supplierToAdd));
+        } else {
```
Comment (qlchan24) [inline src/main/java/seedu/clinic/logic/commands/AddCommand.java]: perhaps its better code quality to explicitly specify the else with a separate if statement.

**Review by qlchan24 — COMMENTED** [review]
Apart from possible minor issues, LGTM

File: src/main/java/seedu/clinic/logic/commands/AddCommand.java (hunk 1 above)
Comment (jeffreytjs) [reply src/main/java/seedu/clinic/logic/commands/AddCommand.java]: Hmmm... Okay I'm including these line breaks,

File: src/main/java/seedu/clinic/logic/commands/AddCommand.java (hunk 3 above)
Comment (jeffreytjs) [reply src/main/java/seedu/clinic/logic/commands/AddCommand.java]: Thanks for pointing it out, I do agree with you and have specified in the next commit!

**Review by qlchan24 — APPROVED** [review]
LGTM

**Pull request merged**